docs/proposals/11-cluster-health.md

```diff
@@ -0,0 +1,298 @@
+            * [Possible Actions](#possible-actions-2)
+      * [Condition Types of future relevance](#condition-types-of-future-relevance)
+         * [InternalNetworkHealthy](#internalnetworkhealthy)
+            * [Possible Actions](#possible-actions-3)
+         * [CloudProviderHealthy](#cloudproviderhealthy)
+            * [Possible Actions](#possible-actions-4)
+         * [CloudProviderQuotaExceeded](#cloudproviderquotaexceeded)
+            * [Possible Actions](#possible-actions-5)
+         * [ScaleThresholdReached](#scalethresholdreached)
+            * [Possible Actions](#possible-actions-6)
+   * [Alternatives](#alternatives)
+      * [Gardener Shoot resource](#gardener-shoot-resource-1)
+      * [Gardener Extensions Cluster Resource](#gardener-extensions-cluster-resource-1)
+      * [Custom Cluster Resource](#custom-cluster-resource)
+## Terminology
+
+### Cluster Health Condition
+
+Any health condition in a Kubernetes cluster that is significant in the scope of the cluster.
+
+For example, the cluster `kube-apiserver` being not reachable is a cluster health condition.
+
+As a counter-example, a single `Node` being `NotReady` may not be a cluster health condition. However, a significant percentage (say, 20%) of the total `nodes` in the cluster being `NotReady` might be cluster health condition indicative of some cluster-wide (or infrastructure) issue.
+
+### Actionable Cluster Health Condition
+
+Any cluster health condition on which some actions can be taken. Such actions could be automated or manual.
+Also, different purposes could be motivating such actions.
+Some such motivations could be as mentioned below.
+- Corrective - remedy the condition and restore the cluster back to health.
+- Alerting - notify or alert the relevant target audience about the cluster health condition.
+- Mitigating - mitigate further (potentially cascading) damage to the cluster while the cluster health condition lasts.
+
+## Motivation
+
+While some of the information about the [cluster health conditions](#cluster-health-condition) for a shoot cluster is published in a [standard way](#gardener-shoot-resource) (in the `Shoot` resource), [actionable cluster health conditions](#actionable-cluster-health-condition) (say some threshold percentage of `nodes` are `NotReady` or that the cluster apiserver is not reachable externally but reachable internally from the seed cluster) are more distributed, implicit and isolated among different extensions and controllers.
+
+Where such information is standardized, it is not uniformly convenient for all potential consuming extensions and controllers (both internal and external to Gardener).
+
+For example,
+- Gardener
+
+  - The [shoot care controlelr](https://github.com/gardener/gardener/blob/master/pkg/gardenlet/controller/shoot/shoot_care_control.go#L160) maintains some important the shoot cluster health conditions in the Gardener `Shoot` [resource](#gardener-shoot-resource).
```

Comment:
s/controlelr/controller/
s/some important the shoot cluster health conditions/some important the cluster health conditions/